### PR TITLE
Node Rows

### PR DESCRIPTION
**Pull request by kylemas**

### CODE REVIEW

**Comment by McTano** [comment]
To answer the question you asked, I'm not sure what the best way is to get the row label to show up at the right height. It might be kind of tricky. To show it once per row, we need to add rows outside of the tree instead of the node div, and then figure out how far apart they need to be vertically spaced.

For now, don't worry about how the labels get displayed. First, let's get the nodes labelled with the correct row. Right now they're not getting labelled correctly.

The point of having a row # on each node is that it is independent of the node's position in the tree. Otherwise we wouldn't actually need it.

Consider this truth tree.
Screenshot from 2019-10-30 12-33-00

There are four nodes in row 3, but the two on the left were added before the two on the right.
Therefore the left side should be in row 3, but the right side should be in row 4.

The forall-x text doesn't seem to include any examples which actually illustrate this point.
Here's another example I found online.
image

**Comment by McTano** [comment]
You'll need to keep track of the current row separately from the state of the tree.

**Comment by kylemas** [comment]
Hmm, okay. I thought of doing them outside of the state of the tree, though was having a bit of difficulty on thinking about the specifics. I'll have a bit more time to work on it this week, so I will try this out.

Also, thanks for pointing out the error with the rows. I actually didn't notice that since I just assumed that it would be on the same row. I will add a fix to this then start working on it getting to appear like how it should be.

**Comment by kylemas** [comment]
@McTano 

I've fixed up how the rows increment now. I kept track of the 'maxRow' outside of the state of the tree and is incremented when splitBranch or continueBranch is called.

File: src/components/App.tsx [hunk 1]
```diff
@@ -35,13 +40,13 @@ const App: React.FC = (): JSX.Element => {
   const handleSubmitPremises = (premises: string) => {
     setPremises(premises)
     setTree(parsePremises(premises.split(',')))
+    setRow(tree.forest.length + 1)
```
Comment (McTano) [inline src/components/App.tsx]: As above, the length of the forest is always 1, so this doesn't quite work.

Try it with the default example. You get two 3s.

Comment (McTano) [reply src/components/App.tsx]: You could try having parsePremises return the number of premises and using that.

File: src/typings/TreeNode.tsx [hunk 2]
```diff
@@ -5,9 +5,11 @@ export interface TreeNode {
   closed: boolean
   rule: string
   id: string
+  row: number
+  split: boolean
```
Comment (McTano) [inline src/typings/TreeNode.tsx]: This didn't get used and should be removed.

**Pull request merged**